docs/index.md

```diff
@@ -31,23 +31,62 @@ Machine on Linux or for installing Machine alone without Docker Toolbox, see the
+driver you want use. The driver represents the virtual environment. For example,
+on a local Linux, Mac, or Windows system the driver is typically Oracle Virtual
+Box. For cloud providers, Docker Machine supports drivers such as AWS, Microsoft
+Azure, Digital Ocean and many more. The Docker Machine reference includes a
+complete [list of the supported drivers](drivers/index.md).
+
+Since Docker runs on Linux, each VM that Docker Machine provisions relies on a
+base operating system. For convenience, there are default base operating
+systems. For the Oracle Virtual Box driver, this base operating system is the
+`boot2docker.iso`. For drivers used to connect to cloud providers, the base
+operating system is Ubuntu 12.04+. You can change this default when you create a
+machine. The Docker Machine reference includes a complete [list of the supported
+operating systems](drivers/os-base.md).
 
 For each machine you create, the Docker host address is the IP address of the
 Linux VM. This address is assigned by the `docker-machine create` subcommand.
 You use the `docker-machine ls` command to list the machines you have created.
-The `docker-machine ip <machine-name>` command returns a specific host's IP address.
+The `docker-machine ip <machine-name>` command returns a specific host's IP
+address.
 
 Before you can run a `docker` command on a machine, you configure your
 command-line to point to that machine. The `docker-machine env <machine-name>`
 subcommand outputs the configuration command you should use. When you run a
 container on the Docker host, the container's ports map to ports on the VM.
 
-For a complete list of the `docker-machine` subcommands, see the [Docker Machine subcommand reference](reference/index.md).
+For a complete list of the `docker-machine` subcommands, see the [Docker Machine
+subcommand reference](reference/index.md).
+
+## Crash Reporting
+
+Provisioning a host is a complex matter that can fail for a lot of reasons.
+Your workstation may have a wide variety of shell, network configuration, VPN,
+proxy or firewall issues.  There are also reasons from the other end of the
+chain: your cloud provider or the network in between.
+
+To help `docker-machine` be as stable as possible, we added a monitoring of
+crashes whenever you try to `create` or `upgrade` a host. This will send, over
+HTTPS, to Bugsnag some information about your `docker-machine` version, build,
+OS, ARCH, the path to your current shell and, the history of the last command as
+you could see it with a `--debug` option.  This data is sent to help us pinpoint
+recurring issues with `docker-machine` and will only be transmitted in the case
+of a crash of `docker-machine`.
+
+If you wish to opt out of error reporting, you can create a `no-error-report`
+file in your `$HOME/.docker/machine` directory, and Docker Machine will disable
+this behavior.  e.g.: 
+
+    $ mkdir -p ~/.docker/machine && touch ~/.docker/machine/no-error-report
```

Comment:
I did add this bit, just so users could easily copy-paste the opt-out command if desired.